src/SFA.DAS.ApplyService.Web/Services/OverallOutcomeAugmentationService.cs

```diff
@@ -0,0 +1,334 @@
+                {
+                    sequencesWithModerationFails.Remove(sequence);
+                }
+            }
+        }
+
+        private static void BuildSequencesWithModerationFails(List<AssessorSequence> sequences,
+            List<AssessorSequence> sequencesWithModerationFails)
+        {
+            foreach (var sequence in sequences)
+            {
+                foreach (var section in sequence.Sections)
+                {
+                    if (section.Pages != null)
+                    {
+                        if (sequencesWithModerationFails.All(x => x.SequenceNumber != sequence.SequenceNumber))
+                            sequencesWithModerationFails.Add(new AssessorSequence
+                            {
+                                SequenceNumber = sequence.SequenceNumber,
+                                SequenceTitle = sequence.SequenceTitle,
+                                Sections = new List<AssessorSection>()
+                            });
+
+                        sequencesWithModerationFails.FirstOrDefault(x => x.SequenceNumber == section.SequenceNumber)
+                            ?.Sections.Add(section);
+                    }
+                }
+            }
+        }
+
+        private static void AddPagesToSequencesFromFailedDetails(List<AssessorSequence> sequences, List<ClarificationPageReviewOutcome> failedDetails)
+        {
+            foreach (var sequence in sequences)
+            {
+                foreach (var section in sequence.Sections)
+                {
+                    foreach (var outcome in failedDetails)
+                    {
+                        if (outcome.SequenceNumber == section.SequenceNumber &&
+                            outcome.SectionNumber == section.SectionNumber)
+                        {
+                            if (section.Pages==null)
+                                section.Pages = new List<Page>();
+
+                            if (section.Pages.All(x => x.PageId == outcome.PageId))
+                            {
+                                section.Pages.Add(new Page {PageId = outcome.PageId, Active = true});
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+
+        private string GetTitleForPage(Page page)
+        {
+            var title = _assessorLookupService.GetTitleForPage(page.PageId);
+            if (string.IsNullOrEmpty(title))
+            {
+                title = _assessorLookupService.GetSectorNameForPage(page.PageId);
+            }
+
+            return title;
+        }
+
+
+        private List<Question> GatherAndTypePageQuestions(Page pageDetails)
+        {
+            foreach (var q in pageDetails.Questions)
+            {
+                if (string.Equals(q.Input.Type, QuestionType.CheckboxList,
+                    StringComparison.CurrentCultureIgnoreCase))
+                    q.Input.Type = QuestionType.CheckboxList;
```

Comment:
Might be better to do this for readability:

QuestionType.XXX.Equals(q.Input.Type, StringComparison.CurrentCultureIgnoreCase)

Reply:
ok